ssl/t1_lib.c

```diff
@@ -1682,22 +1682,25 @@ int SSL_get_shared_sigalgs(SSL *s, int idx,
                            int *psign, int *phash, int *psignhash,
                            unsigned char *rsig, unsigned char *rhash)
 {
-    const SIGALG_LOOKUP *shsigalgs;
     if (s->cert->shared_sigalgs == NULL
         || idx >= (int)s->cert->shared_sigalgslen
         || s->cert->shared_sigalgslen > INT_MAX)
         return 0;
-    shsigalgs = s->cert->shared_sigalgs[idx];
-    if (phash != NULL)
-        *phash = shsigalgs->hash;
-    if (psign != NULL)
-        *psign = shsigalgs->sig;
-    if (psignhash != NULL)
-        *psignhash = shsigalgs->sigandhash;
-    if (rsig != NULL)
-        *rsig = (unsigned char)(shsigalgs->sigalg & 0xff);
-    if (rhash != NULL)
-        *rhash = (unsigned char)((shsigalgs->sigalg >> 8) & 0xff);
+    if (idx >= 0) {
```

Comment:
That is an option, but I was leaning towards this approach since `SSL_get_sigalgs` also has this pattern with `if (idx >= 0)`.

Should I unshift it or is it OK like this?